akka-stream/src/main/scala/akka/stream/impl/io/SslTlsCipherActor.scala

```diff
@@ -299,6 +299,8 @@ private[akka] class SslTlsCipherActor(settings: ActorMaterializerSettings, sslCo
       } catch {
         case ex: SSLException ⇒
           if (tracing) log.debug(s"SSLException during doUnwrap: $ex")
+          fail(ex, closeTransport = false)
+          engine.closeInbound()
```

Comment:
Thanks for the investigation and the explanation, as perverse as it sounds it makes perfect sense to me—meaning it matches my previous experience with SSLEngine.